packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1019,6 +1042,107 @@ class ThemeData extends Diagnosticable {
     properties.add(DiagnosticsProperty<ColorScheme>('colorScheme', colorScheme, defaultValue: defaultData.colorScheme));
     properties.add(DiagnosticsProperty<DialogTheme>('dialogTheme', dialogTheme, defaultValue: defaultData.dialogTheme));
     properties.add(DiagnosticsProperty<Typography>('typography', typography, defaultValue: defaultData.typography));
+    properties.add(DiagnosticsProperty<CupertinoThemeData>('cupertinoOverrideTheme', cupertinoOverrideTheme, defaultValue: defaultData.cupertinoOverrideTheme));
+  }
+}
+
+/// A [CupertinoThemeData] that defers unspecified theme attributes to an
+/// upstream Material [ThemeData].
+///
+/// This type of [CupertinoThemeData] is used when using a Material [Theme]
+/// instead of a [CupertinoTheme] to harmonize Cupertino widgets and an ancestor
+/// Material [Theme]'s [ThemeData].
+///
+/// For a [CupertinoThemeData] whose unspecified attributes default to iOS
```

Comment:
I think you want to explain three scenarios here:
- The CupertinoTheme is based on the MaterialTheme. This is the default and is based on MaterialBasedCupertinoThemeData.
- The CupertinoTheme is based on the MaterialTheme with some overrides per the cupertinoOverrideTheme.
- A CupertinoTheme  widget is added by the app, typically using MaterialApp's builder parameter.  In this case the Material theme and the app's cupertinoOverrideTheme have no effect on the Cupertino widgets.

Reply:
Good point. Done